### PR TITLE
make parseLsListLine support parse another line format.

### PR DESCRIPTION
**Pull request by zcc35357949**
List() method can not parse the response file lines correctly. 
I get the below `ls` result: 
```
drwxrwxrwx      1 testftp    Local Users           84 Oct 16 18:06 zcc
drwxrwxrwx      1 testftp    Local Users            0 Oct 10 17:46 test111
```

### CODE REVIEW

File: parse.go [hunk 1]
```diff
@@ -116,36 +116,39 @@ func parseLsListLine(line string, now time.Time, loc *time.Location) (*Entry, er
...
 	case 'l':
 		e.Type = EntryTypeLink
+		timeBeginIndex, timeEndIndex = fieldsLen-6, fieldsLen-3
 
 		// Split link name and target
-		if i := strings.Index(e.Name, " -> "); i > 0 {
-			e.Target = e.Name[i+4:]
-			e.Name = e.Name[:i]
+		if fields[fieldsLen-2] == "->" {
+			e.Target = fields[fieldsLen-2]
```
Comment (jlaffaye) [inline parse.go:144]: So this would be equal to `->` ?

Comment (zcc35357949) [reply parse.go:144]: > So this would be equal to `->` ?

`scanner.NextFields` has trimed the left and right space

**Review by corny — CHANGES_REQUESTED** [review]
Please add your example to the list of test cases.

**Comment by mafredri** [comment]
Tried out this PR, noticed that it breaks some existing tests and also the FTP server I'm currently working against.

**Pull request closed without merging**